packages/terra-data-grid/tests/jest/FlowsheetDataGrid.test.jsx

```diff
@@ -253,3 +259,416 @@ describe('FlowsheetDataGrid', () => {
+      ...dataFile,
+      rows: [
+        ...dataFile.rows.slice(0, 3),
+        {
+          id: '4',
+          cells: [
+            { content: 'Oxygen Flow Rate (L/min)' },
+            { content: '63', isSelected: true },
+            { content: '47' },
+          ],
+        },
+      ],
+    };
+
+    const wrapper = mountWithIntl(
+      <FlowsheetDataGrid
+        id="test-terra-flowsheet-data-grid"
+        columns={dataFile.cols}
+        rows={updatedDataFile.rows}
+        ariaLabel="Test Flowsheet Data Grid"
+        onCellSelect={mockOnCellSelect}
+      />,
+    );
+
+    const selectableCell = wrapper.find('Row').at(3).find('td.selectable').at(0);
+    selectableCell.simulate('keydown', { keyCode: SPACE_KEY });
+
+    expect(mockOnCellSelect).toHaveBeenCalledWith('4', 'Column-1');
+    expect(mockOnCellRangeSelect).not.toHaveBeenCalled();
+  });
+
+  it('verifies single cell selection does not occur when an unselectable cell is clicked', () => {
+    const updatedDataFile = {
+      ...dataFile,
+      rows: [
+        ...dataFile.rows.slice(0, 3),
+        {
+          id: '4',
+          cells: [
+            { content: 'Oxygen Flow Rate (L/min)' },
+            { content: '63', isSelectable: false },
+            { content: '47' },
+          ],
+        },
+      ],
+    };
+
+    const wrapper = mountWithIntl(
+      <FlowsheetDataGrid
+        id="test-terra-flowsheet-data-grid"
+        columns={dataFile.cols}
+        rows={updatedDataFile.rows}
+        ariaLabel="Test Flowsheet Data Grid"
+        onCellSelect={mockOnCellSelect}
+      />,
+    );
+
+    const selectableCell = wrapper.find('Row').at(3).find('td:not(.selectable)').at(0);
+    selectableCell.simulate('mouseDown');
+
+    expect(mockOnCellSelect).not.toHaveBeenCalled();
+    expect(mockOnCellRangeSelect).not.toHaveBeenCalled();
+  });
+
+  it('verifies single cell selection does not occur when Space is pressed on an unselectable cell', () => {
+    const updatedDataFile = {
+      ...dataFile,
+      rows: [
+        ...dataFile.rows.slice(0, 3),
+        {
+          id: '4',
+          cells: [
+            { content: 'Oxygen Flow Rate (L/min)' },
+            { content: '63', isSelectable: false },
+            { content: '47' },
+          ],
+        },
+      ],
+    };
+
+    const wrapper = mountWithIntl(
+      <FlowsheetDataGrid
+        id="test-terra-flowsheet-data-grid"
+        columns={dataFile.cols}
+        rows={updatedDataFile.rows}
+        ariaLabel="Test Flowsheet Data Grid"
+        onCellSelect={mockOnCellSelect}
+      />,
+    );
+
+    const selectableCell = wrapper.find('Row').at(3).find('td:not(.selectable)').at(0);
+    selectableCell.simulate('keydown', { keyCode: SPACE_KEY });
+
+    expect(mockOnCellSelect).not.toHaveBeenCalled();
+    expect(mockOnCellRangeSelect).not.toHaveBeenCalled();
+  });
+
+  it('verifies Shift+Arrow keys selects a range of cells when an anchor cell is initially clicked', () => {
```

Comment:
Can we break these range selection tests out of the `Single cell selection` test describe module and have them under something like `Multi select` or `Range select` test module.

Reply:
Done